src/charactersheet/viewmodels/character/companions/form.html

```diff
@@ -0,0 +1,355 @@
+            source: $component.prePopFilter,
+            onselect: $component.populate,
+          }
+        "
+      />
+    </div>
+    <div class="form-group col-xs-12">
+      <div class="d-flex justify-content-between">
+        <label class="d-flex-1">
+          Image URL<i class="required"></i></label>
+        </label>
+      </div>
+      <input
+        type="url"
+        class="form-control"
+        name="Image Link"
+        placeholder="http://myurl.com/image.jpg"
+        data-bind="
+          value: sourceUrl,
+          event: { blur: $component.reviewInput, invalid: $component.invalidate },
+          attr: { ...$component.validation.sourceUrl },
+        "
+      />
+    </div>
+    <div class="form-group col-xs-12 col-md-6">
+      <label>Size</label>
+      <input
+        type="text"
+        class="form-control"
+        name="name"
+        placeholder="Gargantuan"
+        data-bind="
+          value: size,
+          event: { blur: $component.reviewInput, invalid: $component.invalidate },
+          attr: { ...$component.validation.size },
+        "
+      />
+    </div>
+    <div class="form-group col-xs-12 col-md-6">
+      <label>Type</label>
+      <input
+        type="text"
+        class="form-control"
+        name="name"
+        placeholder="Dragon"
+        data-bind="
+          value: type,
+          event: { blur: $component.reviewInput, invalid: $component.invalidate },
+          attr: { ...$component.validation.type },
+        "
+      />
+    </div>
+    <div class="form-group col-xs-12 col-md-6">
+      <label>Speed</label>
+      <input
+        type="text"
+        class="form-control"
+        name="speed"
+        placeholder="30ft"
+        data-bind="
+          value: speed,
+          event: { blur: $component.reviewInput, invalid: $component.invalidate },
+          attr: { ...$component.validation.speed },
+        "
+      />
+    </div>
+    <div class="form-group col-xs-12 col-md-6">
+      <label>Alignment</label>
+      <input
+        type="text"
+        class="form-control"
+        name="alignment"
+        placeholder="Chaotic Evil"
+        data-bind="
+          value: alignment,
+          event: { blur: $component.reviewInput, invalid: $component.invalidate },
+          attr: { ...$component.validation.alignment },
+        "
+      />
+    </div>
+  </div>
+  <div class="row">
+    <hr />
+    <div class="form-group col-xs-12 col-lg-4">
+      <label>Armor Class<i class="required"></i></label>
+      <input
+        type="number"
+        class="form-control"
+        name="armorClass"
+        placeholder="22 (natural armor)"
+        data-bind="
+          value: armorClass,
+          event: { blur: $component.reviewInput, invalid: $component.invalidate },
+          attr: { ...$component.validation.armorClass },
+        "
+      />
+    </div>
+    <div class="form-group col-xs-12 col-md-6 col-lg-4">
+      <label>Hit Points<i class="required"></i></label>
```

Comment:
what does `<i class="required">` required do? I don't see anything pop up in the UI.
Furthermore, it doesn't look like this field is actually required. I found that the minimum fields needed to add a companion is `name` and `ac`.

Reply:
I think that maxHitPoints should be required; however, in the spirit of backwards compatibility, I allowed it to use the default of 10 from the api. 

Based on some of the feedback I think I saw in the api, I will re-examine.

Reply:
the required class is a vestibule of the source I copied rom. I've removed them, and updated the required fields in the model (which displays via bootstrap field validation)